registrations/api.py

```diff
@@ -220,8 +221,124 @@ def send_invitation(self, request, pk=None, version=None):
+            queryset = self.filter_by_text(queryset, text_param)
+
+        if status_param := request.query_params.get("attendee_status", None):
+            queryset = self.filter_by_attendee_status(queryset, status_param)
+
+        return queryset
+
+    def filter_by_registration(self, queryset, registration_param):
+        # Get admin organizations and descendants only once instead of using is_admin method
+        admin_organizations = (
+            self.request.user.get_admin_organizations_and_descendants()
+        )
+
+        registrations = []
+        for pk in registration_param.split(","):
+            try:
+                registration = Registration.objects.select_related(
+                    "event__publisher"
+                ).get(pk=pk)
+            except (ValueError, Registration.DoesNotExist):
+                raise ParseError(
+                    _("Registration with id {pk} doesn't exist.").format(pk=pk)
+                )
+
+            if not (
+                self.request.user.is_superuser
+                or registration.registration_user_accesses.filter(
+                    email=self.request.user.email
+                ).exists()
+                or registration.publisher in admin_organizations
+            ):
+                raise DRFPermissionDenied(
+                    _(
+                        "Only the admins of the organization {organization} have access rights."
+                    ).format(organization=registration.publisher)
+                )
+
+            registrations.append(registration)
+        return queryset.filter(registration__in=registrations)
+
+    def filter_by_user(self, queryset, user):
+        # By default return only signups of registrations to which user has admin rights or is registration user
+        return queryset.filter(
+            Q(registration__registration_user_accesses__email=user.email)
+            | Q(
+                registration__event__publisher__in=user.get_admin_organizations_and_descendants()
+            )
+        )
+
+    def _build_text_annotations(self):
+        relation = self._list_filter_relation_name
+        return {
+            f"{relation}first_last_name": Concat(
+                f"{relation}first_name", Value(" "), f"{relation}last_name"
+            ),
+            f"{relation}last_first_name": Concat(
+                f"{relation}last_name", Value(" "), f"{relation}first_name"
+            ),
+        }
+
+    def _build_text_filter(self, text_param):
+        relation = self._list_filter_relation_name
+        filters = {
+            f"{relation}first_last_name__icontains": text_param,
+            f"{relation}last_first_name__icontains": text_param,
+            f"{relation}email__icontains": text_param,
+            f"{relation}extra_info__icontains": text_param,
+            f"{relation}membership_number__icontains": text_param,
+            f"{relation}phone_number__icontains": text_param,
+        }
+        q_set = Q()
```

Comment:
there could be one empty line above this to let the code breath a bit => improves readability

Reply:
Done